packages/nitro-protocol/gas-benchmarks/gas.ts

```diff
@@ -71,28 +73,47 @@ export const gasRequiredTo: GasRequiredTo = {
   },
   ETHexitSad: {
     // Scenario: counterparty goes offline
-    // initially                 ⬛ ->  X  -> 🧑‍🤝‍🧑
-    // challenge + timeout       ⬛ -> (X) -> 🧑‍🤝‍🧑
-    // pushOutcomeAndTransferAll ⬛ -> 🧑‍🤝‍🧑
+    // initially                 ⬛ ->  X  -> 👩
+    // challenge + timeout       ⬛ -> (X) -> 👩
+    // pushOutcomeAndTransferAll ⬛ --------> 👩
     vanillaNitro: {
       challenge: 93404,
       pushOutcomeAndTransferAll: 107742,
       total: 201146,
     },
   },
   ETHexitSadLedgerFunded: {
-    // Scenario: counterparty goes offline
     vanillaNitro: {
-      // initially                   ⬛ ->  L  ->  X  -> 🧑‍🤝‍🧑
-      // challenge X and timeout     ⬛ ->  L  -> (X) -> 🧑‍🤝‍🧑
-      // challenge L and timeout     ⬛ -> (L) -> (X) -> 🧑‍🤝‍🧑
-      // pushOutcomeAndTransferAllL  ⬛ -> (X) -> 🧑‍🤝‍🧑
-      // pushOutcomeAndTransferAllX  ⬛ -> 🧑‍🤝‍🧑
+      // initially                   ⬛ ->  L  ->  X  -> 👩
+      // challenge X, L and timeout  ⬛ -> (L) -> (X) -> 👩
+      // pushOutcomeAndTransferAllL  ⬛ --------> (X) -> 👩
+      // pushOutcomeAndTransferAllX  ⬛ ---------------> 👩
       challengeX: 93404,
-      challengeL: 92122,
+      challengeL: 92338,
       pushOutcomeAndTransferAllL: 58640,
       pushOutcomeAndTransferAllX: 107742,
-      total: 351908,
+      total: 352124,
+    },
+  },
+  ETHexitSadVirtualFunded: {
+    vanillaNitro: {
+      // initially                   ⬛ ->  L  ->  G  ->  J  ->  X  -> 👩
+      // challenge L,G,J,X + timeout ⬛ -> (L) -> (G) -> (J) -> (X) -> 👩
+      // pushOutcomeAndTransferAllL  ⬛ --------> (G) -> (J) -> (X) -> 👩
+      // pushOutcomeG                ⬛ --------> (G) -> (J) -> (X) -> 👩
+      // pushOutcomeJ                ⬛ --------> (G) -> (J) -> (X) -> 👩
+      // claimG                      ⬛ ----------------------> (X) -> 👩
+      // pushOutcomeAndTransferAllX  ⬛ -----------------------------> 👩
+      challengeL: 92350,
+      challengeG: 94645,
+      challengeJ: 101748,
+      challengeX: 93404,
+      pushOutcomeAndTransferAllL: 58652,
+      pushOutcomeG: 61422,
+      pushOutcomeJ: 60558,
+      claimG: 58856,
+      pushOutcomeAndTransferAllX: 107742,
+      total: 729377,
```

Comment:
TLDR it costs at least 729K gas to liquidate a virtually funded channel, in the worst case.

Reply:
This is quite a bit and potentially makes ZK verification costs (~400-600k) look less insane. IF a proof could batch the channel closures somehow (rather than proof-per closure) there could be a substantial saving.

Reply:
Most of these operations would disappear by optimizing:
- [before](https://www.notion.so/statechannels/Virtual-funding-optimizations-16998f6d3baa41a49dd69b9948721b66#380ea45d9b2c4c93abda7dc6734691c8) 7 operations
- [after](https://www.notion.so/statechannels/Virtual-funding-optimizations-16998f6d3baa41a49dd69b9948721b66#4199ddc61f1d495fb6374171381b72cb): 3 operations

**Note**: These numbers do not account for `pushOutcome` operations. `pushOutcome` can be removed by changing the asset holder/adjudicator architecture.

Reply:
This isn't to discount the possibility that batching channel closures can't provide a large improvement!

Batching the closure of many joint channels would indeed help. But, that's a can of worms: if challenges are batched, then is the channel status still visible? Are there new griefing attacks?

Reply:
It looks like we can save a lot of gas with some minor changes, even without the more adventurous VFP optimizations.

 I'm not sure I see a problem with having a `batchChallenge` method? This would simply accept an array of `challenge` calls. That should save at least 3x21K = 63K gas (the base fees).

And it looks like we would save at least another 120K by [deprecating `pushOutcome`](https://www.notion.so/statechannels/Deprecate-discrete-AssetHolders-884d5728716d41ef9fc2d13c5d5d3281). 

Together, a >~25% saving without breaking a sweat.